source/slang/slang-reflection-api.cpp

```diff
@@ -3472,7 +3472,20 @@ SLANG_API char const* spReflectionFunction_GetName(SlangReflectionFunction* inFu
 {
     auto func = convertToFunc(inFunc);
     if (!func)
+    {
+        // If convertToFunc failed, this might be an overloaded function.
+        // Try to get the name from the first overload candidate.
+        auto overloadedFunc = convertToOverloadedFunc(inFunc);
+        if (overloadedFunc && overloadedFunc->lookupResult2.items.getCount() > 0)
+        {
+            auto firstOverload = overloadedFunc->lookupResult2.items[0].declRef;
+            if (auto funcDeclRef = firstOverload.as<FunctionDeclBase>())
+            {
+                return getText(funcDeclRef.getDecl()->getName()).getBuffer();
+            }
+        }
         return nullptr;
+    }
 
     return getText(func.getDecl()->getName()).getBuffer();
```

Comment:
No need for the first nested `if`.
```suggestion
    auto func = convertToFunc(inFunc);
    if (func)
        return getText(func.getDecl()->getName()).getBuffer();
    
    // If convertToFunc failed, this might be an overloaded function.
    // Try to get the name from the first overload candidate.
    auto overloadedFunc = convertToOverloadedFunc(inFunc);
    if (overloadedFunc && overloadedFunc->lookupResult2.items.getCount() > 0)
    {
        auto firstOverload = overloadedFunc->lookupResult2.items[0].declRef;
        if (auto funcDeclRef = firstOverload.as<FunctionDeclBase>())
        {
            return getText(funcDeclRef.getDecl()->getName()).getBuffer();
        }
    }
    return nullptr;
```